devimint/src/external.rs

```diff
@@ -661,6 +668,131 @@ pub async fn open_channel(
+    {
+        return open_channel(process_mgr, bitcoind, cln, lnd).await;
+    }
+
+    tokio::try_join!(
+        gw_cln.wait_for_chain_sync(bitcoind),
+        gw_lnd.wait_for_chain_sync(bitcoind)
+    )?;
+    info!(target: LOG_DEVIMINT, "block sync done");
+    let cln_addr = gw_cln.get_funding_address().await?;
+
+    bitcoind.send_to(cln_addr, 100_000_000).await?;
+    bitcoind.mine_blocks(10).await?;
+
+    let lnd_pubkey = gw_lnd.lightning_pubkey().await?;
+    let cln_pubkey = gw_cln.lightning_pubkey().await?;
+
+    gw_cln
+        .connect_to_peer(
+            lnd_pubkey.parse()?,
+            format!("{}:{}", "127.0.0.1", process_mgr.globals.FM_PORT_LND_LISTEN),
+        )
+        .await?;
+
+    gw_cln
+        .open_channel(lnd_pubkey.clone(), 10_000_000, Some(5_000_000))
+        .await?;
+
+    poll("list peers", || async {
+        let num_peers = cln
+            .request(cln_rpc::model::requests::ListpeersRequest {
+                id: Some(
+                    lnd_pubkey
+                        .parse()
+                        .context("parse lnd pubkey")
+                        .map_err(ControlFlow::Break)?,
+                ),
+                level: None,
+            })
+            .await
+            .map_err(ControlFlow::Break)?
+            .peers
+            .len();
+        poll_eq!(num_peers, 1)
+    })
+    .await?;
+    bitcoind.mine_blocks(10).await?;
+
+    poll("Wait for channel update", || async {
+        let mut lnd_client = lnd.client.lock().await;
+        let channels = lnd_client
+            .lightning()
+            .list_channels(ListChannelsRequest {
+                active_only: true,
+                ..Default::default()
+            })
+            .await
+            .context("lnd list channels")
+            .map_err(ControlFlow::Break)?
+            .into_inner();
+
+        if let Some(channel) = channels
+            .channels
+            .iter()
+            .find(|channel| channel.remote_pubkey == cln_pubkey)
+        {
+            let chan_info = lnd_client
+                .lightning()
+                .get_chan_info(ChanInfoRequest {
+                    chan_id: channel.chan_id,
+                })
+                .await;
+
+            match chan_info {
+                Ok(info) => {
+                    let edge = info.into_inner();
+                    if edge.node1_policy.is_some() {
+                        return Ok(());
+                    } else {
+                        warn!(?edge, "Empty chan info");
+                    }
+                }
+                Err(e) => {
+                    warn!(%e, "Getting chan info failed")
+                }
+            }
+        }
+
+        Err(ControlFlow::Continue(anyhow!("channel not found")))
+    })
+    .await?;
+
+    lnd.client
+        .lock()
+        .await
+        .lightning()
+        .update_channel_policy(PolicyUpdateRequest {
+            min_htlc_msat: 1,
```

Comment:
This is necessary for allowing < 1 sat payments right?

Reply:
Yep! Linking #2827